### DISCUSSION TITLE
How to do a incremental/partial update with LexicalOnChangePlugin with restful api?

### DISCUSSION THREAD

**Discussion by GyokuryuTo** [facebook/lexical#4042] [Q&A]
```tsx
<OnChangePlugin
    ignoreSelectionChange
    onChange={(editorState) => {
      console.log(editorState);
      console.log(editorState.toJSON());
    }}
/>
```

I want to create an editor like Notion to realtime update current changes (not the full state) to database.

It seems I can only get the full editorState object in the onChange event, so it's impossible to do a incremental update in this event?
Also `NodeKey `is not included in `editorState.toJSON()` result, so it's not possible to work with incremental update?

I'm also looking at state update listener and mutation listeners, but there is no detailed document with these features.
https://lexical.dev/docs/concepts/editor-state#state-update-listener
https://lexical.dev/docs/concepts/listeners#registermutationlistener

For example, if I do update like this gif, what will happen with these listeners, and how can I merge these changes into one object so I can send this object to my restful api?
![image](https://files.catbox.moe/d1cyey.gif)

**Comment by JohnGarrison78** [comment]
Did you figure this out? It's my use case too, and I'm just exploring the various text editor options. Quill's delta format would appear to make it easy, but that project's essentially dead. Lexical has a history stack. Perhaps there's a way to use the redo items as a lit of changes for the server to apply? Alternatively, a diff could be created from the previous state and the new one. But it seems like the type of data that would already exist.

**Comment by moy2010** [comment]
Making the client the source of truth sounds dangerous to me, specially if you are considering a collaborative feature later on, in which case it's better to use CRDTs.

If you will ever only have one user per editor (a.k.a. don't care about the collaborative use case), then you can simply diff the two states' JSON using JSON diff.
